src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/EgressTests.cs

```diff
@@ -226,6 +267,121 @@ public async Task SharedConcurrencyLimitTest()
                 });
         }
 
+        [Fact]
+        public async Task HttpEgressCancelTest()
+        {
+            await ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                DiagnosticPortConnectionMode.Connect,
+                TestAppScenarios.AsyncWait.Name,
+                appValidate: async (appRunner, apiClient) =>
+                {
+                    int processId = await appRunner.ProcessIdTask;
+
+                    using ResponseStreamHolder responseBox = await apiClient.HttpEgressTraceAsync(processId, durationSeconds: -1);
+
+                    Uri operationUri = responseBox.Response.Headers.Location;
+                    Assert.NotNull(operationUri);
+
+                    // Start consuming the stream
+                    Task drainResponseTask = responseBox.Stream.CopyToAsync(Stream.Null);
+
+                    // Make sure the operation exists
+                    OperationStatusResponse operationResult = await apiClient.GetOperationStatus(operationUri);
+                    Assert.Equal(HttpStatusCode.OK, operationResult.StatusCode);
+                    Assert.True(operationResult.OperationStatus.Status == OperationState.Running);
+
+                    // Cancel the trace operation
+                    HttpStatusCode deleteStatus = await apiClient.CancelEgressOperation(operationUri);
+                    Assert.Equal(HttpStatusCode.OK, deleteStatus);
+
+                    operationResult = await apiClient.GetOperationStatus(operationUri);
+                    Assert.Equal(HttpStatusCode.OK, operationResult.StatusCode);
+                    Assert.Equal(OperationState.Cancelled, operationResult.OperationStatus.Status);
+
+                    await Assert.ThrowsAsync<IOException>(async () => await drainResponseTask);
```

Comment:
```suggestion
                    await Assert.ThrowsAsync<IOException>(() => drainResponseTask);
```

In general, if a function is of the form `async (...) => await ...`, then the `async` and `await` are not necessary; these create a state machine object that is not needed. The exception is if some kind of disposal/unregistration needs to occur before the function returns.